src/bin/pg_dump/dumpall_babel_utils.c

```diff
@@ -45,7 +70,7 @@ isBabelfishDatabase(PGconn *conn)
 	{
 		PGresult *res;
 		int		 ntups;
-		res = PQexec(conn, "SELECT extname FROM pg_extension WHERE extname = 'babelfishpg_tsql';");
+		res = executeQuery(conn, "SELECT extname FROM pg_extension WHERE extname = 'babelfishpg_tsql';");
```

Comment:
why will such a failure happen? why do we want to fail the utility? this code is also applicable to non-babelfish? If yes, we should assume bbf_status = OFF in such cases

Reply:
With this query only we will figure out if the server is a Babelfish or not. The `executeQuery` is a general wrapper for executing a query and handling the error condition. It's better to fail the utility in case server does not return a response (a general case) as we are otherwise probably going to fail at another place due to server not responding correctly.